### PR TITLE
feat: Implement graph for dozer-live ui

### PR DESCRIPTION
**Pull request by supergi0**

### CODE REVIEW

File: dozer-cli/src/live/state.rs [hunk 1]
```diff
@@ -1,181 +1,206 @@
...
             .await
-            .map_err(|e| LiveError::BuildError(Box::new(e)))
+            .map_err(|e| LiveError::BoxedError(Box::new(e)))
     }
 
-    pub fn generate_dot(&self) -> Result<DotResponse, LiveError> {
-        let dozer = self.get_dozer()?;
-        generate_dot(dozer).map_err(|e| LiveError::BuildError(Box::new(e)))
+    pub fn get_graph_schemas(&self) -> Result<SchemasResponse, LiveError> {
+        let dozer = self
```
[the remark this answers is not in the document]
Comment (v3g42) [reply dozer-cli/src/live/state.rs:157]: Yes makes sense.

[the remark this answers is not in the document]
Comment (v3g42) [reply dozer-cli/src/live/state.rs:157]: This is a mistake. Build should only happen on change

File: dozer-cli/src/cli/types.rs [hunk 2]
```diff
@@ -22,6 +22,8 @@ pub struct Cli {
     pub config_paths: Vec<String>,
     #[arg(global = true, long, hide = true)]
     pub config_token: Option<String>,
+    #[arg(global = true, short = 'p', long = "enable-progress")]
+    pub enable_progress: Option<Option<String>>,
```
[the remark this answers is not in the document]
Comment (v3g42) [reply dozer-cli/src/cli/types.rs:26]: couldnt get bool easily with clap. so this is sort of a work around. if you have a better way go ahead. Basically we ll need to read string from the cli and aprse as bool.

File: dozer-cli/src/live/errors.rs [hunk 3]
```diff
@@ -28,4 +28,7 @@ pub enum LiveError {
 
     #[error(transparent)]
     PipelineError(#[from] PipelineError),
+
+    #[error(transparent)]
+    BoxedError(#[from] BoxedError),
```
[the remark this answers is not in the document]
Comment (v3g42) [reply dozer-cli/src/live/errors.rs:33]: clippy was complaining error context was too big. so boxed to get around that warning.

File: dozer-cli/src/live/graph.rs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    id: u32,
+    label: String,
+}
+impl Display for Node {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.label)
+    }
+}
+
+pub fn transform_dag_ui(input_dag: &Dag<SchemaSQLContext>) -> daggy::Dag<Node, &str> {
```
[the remark this answers is not in the document]
Comment (v3g42) [reply dozer-cli/src/live/graph.rs:31]: this is to replace teh current implementation. Simply using Dot format as a transport mechanism. Our current cloud version was a hack i put together a long time ago.

[the remark this answers is not in the document]
Comment (v3g42) [reply dozer-cli/src/live/graph.rs:31]: this builds a `daggy::Dag` that in theory could be converted into other graph formats.

File: dozer-cli/src/live/errors.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (v3g42) [reply dozer-cli/src/live/errors.rs:33]: specifically on `SendError`

File: dozer-cli/src/cli/types.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (chubei) [reply dozer-cli/src/cli/types.rs:26]: `enable_progress: bool` should work. `clap` should recognize `bool` as a flag argument. If run without `-p`, it's false. If run with `-p` it's true. @supergi0 can you make the change and test it?

[the remark this answers is not in the document]
Comment (chubei) [reply dozer-cli/src/cli/types.rs:26]: Quite sure `-p` will work. Using it in at least two other places.

File: dozer-cli/src/live/errors.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (chubei) [reply dozer-cli/src/live/errors.rs:33]: We can consider using `CannotBroadcastConnectResponse<Box<SendError<Result<ConnectResponse, tonic::Status>>>>` if the `SendError` is too big. But given that `ConnectResponse` is just two `Option`s, it's probably some other error types being big?

**Pull request closed without merging**